src/app/components/WorkDetail/DefinitionList.jsx

```diff
@@ -54,15 +57,34 @@ export const DefinitionList = (props) => {
    * @return {string|null}
    */
   const parseEntries = (type, entries) => {
+    let list = [...entries];
+    if (type === 'instances') {
+      list = list.map((item) => {
+        let publisher = item && item.agents && item.agents.find(i => i.roles.indexOf('publisher') > -1);
+        publisher = publisher && publisher.name;
+        return Object.assign({}, item, { publisher });
+      });
+    }
     switch (type) {
-      case 'entities':
+      case 'language':
+        return (
+          <ul>
+            {list.map((entity, i) => (
+              <li key={i.toString()}>
+                {entity.language}
+              </li>
+            ))}
+          </ul>
+        );
+      case 'agents':
         return (
           <ul>
-            {entries.map((entity, i) => (
+            {list.map((entity, i) => (
               <li key={i.toString()}>
                 <Link
                   onClick={event => newSearchRequest(event, `\"${entity.name}\"`, 'author')}
-                  to={{ pathname: '/search', query: { q: `\"${entity.name}\"`, field: 'author' } }}>{entity.name}, {entity.role}
+                  to={{ pathname: '/search', query: { q: `\"${entity.name}\"`, field: 'author' } }}
+                >{entity.name}, {entity.roles.map(role => role)}
```

Comment:
Eventually these roles should be separated by a comma.

Reply:
done